### ISSUE TITLE
Feature Request - Enhanced Unsaved Files Indicator in Cursor

### ISSUE THREAD

**Issue by BhalliBhai** [cursor/cursor#1138]
### **Feature Request - Enhanced Unsaved Files Indicator in Cursor**

Dear Cursor Development Team,

I trust this message finds you well. As a dedicated user of your cursor application, I've been consistently impressed with its efficiency and user-friendly interface. I'm writing to suggest a feature that could further enhance the user experience, particularly for those working on larger projects. Additionally, I'd like to express my gratitude to the Team Cursor who has made this application possible.

**Enhanced Unsaved Files Indicator:**

I propose the implementation of an "_Unsaved Files Indicator_," akin to the functionality available in VS Code, to provide users with a clear overview of pending changes in their project. This feature would include:

**Unsaved Files Count Display**: A numerical badge on the files icon, dynamically reflecting the number of files in the project with unsaved changes.

**Hover Information**: When hovering over the unsaved files count, a tooltip would appear, displaying the names of the files requiring attention, facilitating quick identification and navigation.

**Visual Distinction**: Consideration of a distinctive color or icon overlay on the files icon to ensure immediate visibility of unsaved changes.

**Team Cursor**:

_In addition to the feature request, I'd like to take a moment to express my sincere appreciation for the hard work and dedication of Team Cursor. Your collective efforts have created a tool that significantly contributes to the efficiency and joy of coding for countless users. Including a feature that acknowledges and appreciates Team Cursor within the application would be a wonderful way to recognize your invaluable contributions._

Whether through a dedicated section in the application's about page, a periodic "Team Spotlight," or another suitable method, acknowledging the team behind the scenes would foster a sense of community and gratitude among users.

I believe that combining the proposed Unsaved Files Indicator with a feature recognizing Team Cursor would not only elevate the functionality of your cursor application but also strengthen the sense of collaboration and community surrounding it.

Thank you for considering these suggestions, and most importantly, thank you to Team Cursor for your hard work and dedication.

Best regards,

**_Bhalli_**

**Reference:-**
VS Code: 
![image](https://github.com/getcursor/cursor/assets/94342592/d8249059-3671-48dc-85be-bd9ef201304a)

Cursor: 
![image](https://github.com/getcursor/cursor/assets/94342592/5a2cc6a4-f914-4fc4-8d9e-01c4af5753e9)

**Comment by Med5KDB** [comment]
It seems Cursor changed the position of the activity bar from horizontal to vertical like in VS Code. I prefer the horizontal activity bar as we can see in @BhalliBhai second attached image. Why not keeping it and just show unsaved files indicator?

**Comment by HelenaSanchez167** [comment]
Adding to that, please add the Uncommited Files indicator too. It's my only gripe with the wonderful tool that is Cursor and the only reason I would still use VSCode.

**Comment by raphaelmans** [comment]
> Adding to that, please add the Uncommited Files indicator too. It's my only gripe with the wonderful tool that is Cursor and the only reason I would still use VSCode.

This would be really helpful, there is really a disconnect in experience, which bothers me.

**Comment by Motorratte** [comment]
Yes PLS add this feature. There seems to be no feedback for unsafed changes which is also helpfull to avoid unwanted changes.

**Comment by PoorvangRaapyd** [comment]
@BhalliBhai  @HelenaSanchezBasetis  
You can change to VSC’s own layout (which has the bubble icon) by changing the workbench.activityBar.orientation setting to vertical. From there, you can still switch to a horizontal layout by right-clicking on the activity bar.
